### PR TITLE
moving maven plugin to use agent-instrumeneter

### LINKED ISSUE #156: Maven plugin for static instrumentation using Java agent
Dear @open-telemetry/java-contrib-maintainers I'd like to contribute a Maven plugin, capable of ahead-of-time Java application instrumentation using Java OTel agent (aka "static instrumentation").

The goal of the project is:
- implement proof of concept how existing agent can be integrated into the Maven build cycle to instrument applications during build time
- provide a solution for all use cases where dynamic instrumentation is not feasible (primary example - AWS lambda suffering cold start performance issues with dynamic instrumentation)
- measure startup performance dynamic vs static proving the point

Future goal within the community would be to improve the plugin from PoC into real product shape.

The code has been developed by a group of AGH (university in Krakow, Poland) students for their BSc thesis project (3rd year of studies), with active help from Splunk side.

The code can be found here: https://github.com/agh-static-x/Maven-Plugin

### PR DESCRIPTION
**Pull request by kubawach**
**Description:**

Maven plugin now using instrumented OTel agent from agent-instrumenter module.

**Existing Issue(s):**

Closes #156

**Testing:**

Unit / integration tests

### CODE REVIEW

File: static-instrumenter/maven-plugin/build.gradle.kts [hunk 1]
```diff
@@ -16,13 +18,23 @@ dependencies {
   testImplementation("org.apache.maven.plugin-tools:maven-plugin-annotations:3.6.0")
   testImplementation("org.apache.maven:maven-core:3.5.0")
   testImplementation("org.slf4j:slf4j-simple")
+
+  instrumentedAgent(project(":static-instrumenter:agent-instrumenter", "shadow"))
+}
+
+task<Copy>("copyAgent") {
```
Comment (anuraaga) [inline static-instrumenter/maven-plugin/build.gradle.kts]: Gradle can have problems when multiple tasks work on the same directory, in this case `processResources` and `copyAgent`. I pushed a commit to use `processResources` for this

**Review by anuraaga — APPROVED** [review]

File: static-instrumenter/maven-plugin/build.gradle.kts (hunk 1 above)
Comment (kubawach) [reply static-instrumenter/maven-plugin/build.gradle.kts]: Thanks! Gradlu-fu is strong with you! ;)

File: settings.gradle.kts [hunk 2]
```diff
@@ -9,6 +9,8 @@ pluginManagement {
 
 plugins {
   id("com.gradle.enterprise")
+
+  id("com.github.johnrengelman.shadow") apply false
```
Comment (mateuszrzeszutek) [inline settings.gradle.kts:13]: Why is it needed in the main `settings.gradle.kts` file? Can you add it to the maven plugin `build.gradle.kts` file instead?

**Review by mateuszrzeszutek — APPROVED** [review]

File: settings.gradle.kts (hunk 2 above)
Comment (kubawach) [reply settings.gradle.kts:13]: It's a question to @anuraaga I presume .

Comment (anuraaga) [reply settings.gradle.kts:13]: Sorry yeah I think that would work

**Review by Enkelian — APPROVED** [review]

**Comment by kubawach** [comment]
Any chances of getting this merged?

**Comment by mateuszrzeszutek** [comment]
> Any chances of getting this merged?

Yeah, can you fix the `settings.gradle.kts` comment?

**Comment by trask** [comment]
thanks @kubawach ❤️

**Pull request merged**